pkg/process/util/address.go

```diff
@@ -9,28 +9,48 @@ import (
 	"encoding/binary"
 	"net"
 	"sync"
+
+	"inet.af/netaddr"
 )
 
 // Address is an IP abstraction that is family (v4/v6) agnostic
-type Address interface {
-	Bytes() []byte
-	WriteTo([]byte) int
-	String() string
-	IsLoopback() bool
-	Len() int
+type Address struct {
```

Comment:
I think you could do `type Address netaddr.IP`. I'm actually not sure what the difference between the two would be.

Reply:
You need to embed `netaddr.IP` if you want to expose its methods.
See https://go.dev/play/p/8IoKQMvoZ2K

Reply:
Ah right.